### DISCUSSION TITLE
Kafkahq OIDC Hybrid flow

### DISCUSSION THREAD

**Discussion by swapnilborade** [tchiotludo/akhq#712] [Q&A]
Does Kafahq support hybrid OIDC flow with response_type=code id_token?

**Comment by tchiotludo** [comment]
I think it's handle by micronaut : https://micronaut-projects.github.io/micronaut-security/latest/guide/#io.micronaut.security.oauth2.configuration.OauthClientConfigurationProperties.OpenIdClientConfigurationProperties.AuthorizationEndpointConfigurationProperties
I don't know in detail but I think it's `micronaut.security.oauth2.clients.*.openid.authorization.response-type` that will fit your need

**Reply by swapnilborade** [reply]
Thanks. I tried configuration but it seems that it only supports code or id_token or token ..

https://micronaut-projects.github.io/micronaut-security/latest/api/io/micronaut/security/oauth2/endpoint/authorization/request/ResponseType.html


I got below exception when I tried hybrid "code id_token"

Message: Error resolving property value [micronaut.security.oauth2.clients.oidc.openid.authorization.response-type]. Unable to convert value [code id_token] to target type [ResponseType] due to: No enum constant io.micronaut.security.oauth2.endpoint.authorization.request.ResponseType.code id_token
Path Taken: OauthClientConfigurationProperties.setOpenid([OpenIdClientConfigurationProperties openid]) --> OpenIdClientConfigurationProperties.setAuthorization([AuthorizationEndpointConfigurationProperties authorization]) --> AuthorizationEndpointConfigurationProperties.setResponseType([ResponseType responseType])

**Reply by tchiotludo** [reply]
So you will need to report the issue to micronaut : https://github.com/micronaut-projects/micronaut-security/issues
